tests/runtime/servicers/test_global_predict_servicer_impl.py

```diff
@@ -68,6 +68,31 @@ def test_calling_predict_should_raise_if_module_raises(
     assert "Unhandled exception during prediction" in context.value.message
 
 
+def test_predict_raises_with_grpc_errors(
+    sample_inference_service,
+    sample_predict_servicer,
+    sample_task_model_id,
+    sample_task_unary_rpc,
+):
+    with pytest.raises(CaikitRuntimeException) as context:
+        # SampleModules will raise a RuntimeError if the throw flag is set
+        predict_class = get_inference_request(SampleModule.TASK_CLASS)
+        print(SampleModule.TASK_CLASS)
+
+        request = predict_class(
+            sample_input=HAPPY_PATH_INPUT_DM,
+            throw=True,
+            error="GRPC_RESOURCE_EXHAUSTED",
```

Comment:
There are a lot of things I don't like about mocks but... sometimes I wish we had just used mocks for these 🙃 

It would make it a bit easier to say like
```
with patch(SampleModule, "run") as mock_run:
    mock_run.side_effect = grpc.RpcError(grpc.StatusCode.UNIMPLEMENTED, "some error message")
```
and be able to test out any exception type and content

Not super necessary to change but that might make our tests a little easier to read as well

Reply:
That said- if we have the eventual goal of making tests parallel I don't actually know if patching like that is thread safe. If it's not, then maybe we should keep as is 🤔

Reply:
hmm, good question. Not sure how patching works with parallel tests and specially when there is threading etc that can happen as a result of some of these tests 🤔

Reply:
internet seems to suggest its not: https://stackoverflow.com/questions/26876438/does-pythons-unittest-mock-patch-mutate-global-state#:~:text=11-,mock.,assignment%20statement%20at%20the%20end.

Reply:
> Typically, it's used to modify attributes of modules, which are global state. When used this way, it is not thread safe.

Yeah, sounds like we'd need to patch a specific model instance for the test, not the full module :/